opentelemetry-sdk/src/opentelemetry/sdk/_metrics/view.py

```diff
@@ -0,0 +1,154 @@
+
+        1. Match instruments: When an instrument matches a view, measurements
+           received by that instrument will be processed.
+        2. Customize metric streams: A metric stream is identified by a match
+           between a view and an instrument and a set of attributes. The metric
+           stream can be customized by certain attributes of the corresponding
+           view.
+
+        The attributes documented next serve one of the previous two purposes.
+
+        Args:
+            instrument_type: This is an instrument matching attribute: the
+                class the instrument must be to match the view.
+
+            instrument_name: This is an instrument matching attribute: the name
+                the instrument must have to match the view.
+
+            meter_name: This is an instrument matching attribute: the name
+                the instrument meter must have to match the view.
+
+            meter_version : This is an instrument matching attribute: the
+                version the instrument meter must have to match the view.
+
+            meter_schema URL : This is an instrument matching attribute: the
+                schema URL the instrument meter must have to match the view.
+
+            name: This is a metric stream customizing attribute: the name of
+                the metric stream. If `None`, the name of the instrument will
+                be used.
+
+            description: This is a metric stream customizing attribute: the
+                description of the metric stream. If `None`, the description of
+                the instrument will be used.
+
+            attribute_keys: This is a metric stream customizing attribute: this
+                is a set of attribute keys. If not `None` then only the
+                measurement attributes that are in `attribute_keys` will be
+                used to identify the metric stream.
+
+            aggregation: This is a metric stream customizing attribute: every
+                metric stream has an aggregation instance, this is the class of
+                aggregation this instance will be. If `None` the default
+                aggregation class of the instrument will be used.
+        """
+
+        if (
+            instrument_type
+            is instrument_name
+            is meter_name
+            is meter_version
+            is meter_schema_url
+            is None
+        ):
+            raise Exception(
+                "Some instrument selection "
+                "criteria must be provided for View {name}"
+            )
+
+        self._name = name
+        self._instrument_type = instrument_type
+        self._instrument_name = instrument_name
+        self._meter_name = meter_name
+        self._meter_version = meter_version
+        self._meter_schema_url = meter_schema_url
+
+        self._description = description
+        self._attribute_keys = attribute_keys
+        self._aggregation = aggregation
```

Comment:
I know I've brought this up a few times, but it is very strange to make these all protected and then do `view._foo` everywhere inside the SDK. I think at least `name` and `aggregation` should be public as they need to accessed elsewhere. The other properties should only be used in the match method.

Reply:
The user does not need access to the `name` and `aggregation` attribute, better to keep them private and minimize our API.

Reply:
Can you add a comment saying these will be used internally? It's very confusing when reading the code if there is no notion of encapsulation

Reply:
right, fixed